relay/workflow.py

```diff
@@ -585,16 +585,16 @@ async def handle(self, websocket):
 
         try:
             async for m in websocket:
-                 # TODO: restore after PE-17571
-                # self.logger.debug(f'recv: {m}')
+                # TODO: restore after PE-17571
+                # self.logger.debug(f'recv raw: {m}')
                 e = self.fromJson(m)
                 self.logger.debug(f'recv: {e}')
 
                 _id = e.get('_id', None)
                 _type = e.get('_type', None)
                 request_id = e.get('request_id', None)
-    
-                if _id:
+
+                if _id in self.id_futures.keys():
```

Comment:
could rewrite this as:

fut = self.id_futures.pop(_id, None)
if fut:
   fut.set_result(e)

else:
   handled = False
   ...